### DISCUSSION TITLE
Merging topics without the documents

### DISCUSSION THREAD

**Discussion by maticar92** [MaartenGr/BERTopic#1867] [Q&A]
Hello, Maarten/ Hello All,

I am having a bit of an issue to merge topics. I have a set of models that I've merged to achieve an incremental learning topic pipeline where every month I train a model a join it to the previous month model. 

The problem comes when I need to load a model (saved with safe tensors) and I need to merge a topic in the merged model.

The merge_topics() method required documents as an input :
```
    def merge_topics(self,
                     docs: List[str],
                     topics_to_merge: List[Union[Iterable[int], int]],
                     images: List[str] = None) -> None:
```

I no longer have the documents used to train the previous models accessible and I think if I load them I would not have the order correctly. Is there a workaround to this?

**Comment by MaartenGr** [comment]
Thanks for sharing this. Although it is currently not implemented a (perhaps naive) solution would be to identify which topics should be merged, for instance, topics 1, 2, 3, and simply ignore topics 2 and 3 during the assignment. 

When you want to merge topics without having access to the documents, what you essentially are doing is simply removing topics (2, 3) and re-assigning them a topic (1) without actually updating the topic representation. Sure, you can still update the topic representation using an LLM for this but without the underlying documents methods like c-TF-IDF will not (or be difficult to) work. 

Thus, you can use `calculate_probabilities=True` to generate the probabilities and simply take the max probability and re-assign based on the mapping that you created.
